### ISSUE TITLE
@webpack-blocks/babel: Allow setting presets/plugins

### ISSUE THREAD

**Issue by andywer** [andywer/webpack-blocks#8]
Right now the user is supposed to set the babel config in `package.json` or `.babelrc` file. But especially for the planned preset functionality (#7) it is desirable to optionally be able to provide the babel config from within the webpack config (as create-react-app does it).

# To Do
Set babel-loader's query params if `presets`/`plugins` are passed to the webpack block: https://github.com/babel/babel-loader#usage

**Comment by eXon** [comment]
@andywer what if any block could specify babel query parameters? if we switch the babel block to be added on the post-hook, we could probably do that by using the context just like the mediaType. What do you think?

**Comment by eXon** [comment]
We could even add a block `addBabelPlugin()` or `addBabelPreset()`. But the most obvious use case for me is having something like a flowType() block that would add the flow-type babel plugin within the block.

**Comment by andywer** [comment]
@exon The flow-type use case is quite a good point... 👍

**Comment by andywer** [comment]
Just recognized there is one slight issue: You could set a different `exclude` option on any `babel()` block call...

Proposal: For the sake of simplicity we just take the first call's `exclude` option (or the default if not present).

**Comment by andywer** [comment]
I think I might have found a better solution: [babel6/index.js in PR](https://github.com/andywer/webpack-blocks/pull/45/files#diff-2a6399063cd3f6cc0cfb72239de90a8e)

**Comment by eXon** [comment]
👍  looks like a perfect solution

**Comment by mjsisley** [comment]
Great work on this library!

I think `include` should be an included option for babel(). It allows for white-listing of dependencies for tree-shaking... which is simpler than excluding everything or a regex. 

* https://github.com/developit/preact-compat/issues/155
* https://github.com/rollup/rollup/wiki/jsnext:main

It should probably only allow for one of the two... right now I'm overriding exclude with include

```
const rule = options.include || options.exclude || /\/node_modules\//

  return (fileTypes) => ({
    module: {
      loaders: [
        {
          test: fileTypes('application/javascript'),
          [options.include ? 'include' : 'exclude' ]: Array.isArray(rule) ? rule : [ rule ],
          loaders: [ 'babel-loader?cacheDirectory' ]
        }
      ]
    }
  })
```

**Comment by andywer** [comment]
Thanks, @mjsisley.

Good catch. But I think allowing `include` and `exclude` at the same time should be fine, since webpack allows its loaders to set both. I also know people who actually use both, even though it might not be best practice.

Will update the PR :)

**Comment by andywer** [comment]
Opened PR: #53

**Comment by andywer** [comment]
Published in version 0.3.0.
